src/readcalls.c

```diff
@@ -141,55 +143,61 @@ int readcalls(const char *logfile, bool interactive) {
 
     bool log_changed = false;
 
-    while (fgets(inputbuffer, sizeof(inputbuffer), fp) != NULL) {
+    while ((read = getline(&inputbuffer, &inputbuffer_len, fp)) != -1) {
+	if (inputbuffer_len > 0) {
+	    if (errno == ENOMEM) {
+		fprintf(stderr, "Error in: %s:%d", __FILE__, __LINE__);
+		perror("RuntimeError: ");
+		exit(EXIT_FAILURE);
+	    }
+	    // drop trailing newline
+	    inputbuffer[inputbuffer_len - 1] = '\0';
```

Comment:
`inputbuffer_len` is the allocated buffer size, not the read line size.
Revert this line and then `test_readcalls.c` also.